extensions/amp-subscriptions/0.1/amp-subscriptions.js

```diff
@@ -301,41 +300,71 @@ export class SubscriptionService {
         return this.platformStore_.resolveEntitlement('local',
             Entitlement.empty('local'));
       }
-      const decodedData = this.jwtHelper_.decode(authData);
-      user().assert(
-          decodedData['aud'] == origin ||
-          decodedData['aud'] == sourceOrigin,
+
+      return this.verifyAuthToken_(authData).then(entitlement => {
```

Comment:
Ideally, the `if (!authData) {...}` above should be factored inside `verifyAuthToken_`. This way you will have a single code path and should be easier to add behavior for grantState later.